File: clinica SPRING boot/demo/src/main/java/com/example/clinica/dtos/GetMascotaDto.java
package com.example.clinica.dtos;

import java.util.Date;
import com.example.clinica.entities.Mascota.Sexo;
import com.example.clinica.entities.Mascota.Tamano;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class GetMascotaDto {
    private Long id;
    private String nombre;
    private String especie;
    private String raza;
    private Date fechaNacimiento;
    private Tamano tamano;
    private Double peso;
    private Sexo sexo;
    private String usuarioDni;
    private String imagenUrl;
}

File: clinica_app/lib/pages/data/GetMascotaDto.dart
class GetMascotaDto {
  final int id;
  final String nombre;
  final String especie;
  final String raza;
  final String fechaNacimiento;
  final String sexo;
  final String tamano;
  final double peso;
  final String usuarioDni;
  final String? imagenUrl;

  GetMascotaDto({
    required this.id,
    required this.nombre,
    required this.especie,
    required this.raza,
    required this.fechaNacimiento,
    required this.sexo,
    required this.tamano,
    required this.peso,
    required this.usuarioDni,
    this.imagenUrl
  });

  factory GetMascotaDto.fromJson(Map<String, dynamic> json) {
    return GetMascotaDto(
      id: json['id'],
      nombre: json['nombre'],
      especie: json['especie'],
      raza: json['raza'],
      fechaNacimiento: json['fechaNacimiento'],
      sexo: json['sexo'],
      tamano: json['tamano'],
      peso: (json['peso'] as num).toDouble(),
      usuarioDni: json['usuarioDni'],
      imagenUrl: json['imagenUrl'],
      
    );
  }
}
